tests/Integration/Behaviour/Features/Scenario/CartRule/set_product_restrictions.feature

```diff
@@ -0,0 +1,267 @@
+      | products   | product3          |
+      | attributes | S,M               |
+      | categories | clothes,men       |
+      | categories | women             |
+    And the cart rule restriction group "free_nr_2" should have the following rules:
+      | type          | references                          |
+      | suppliers     | fashionSupplier,accessoriesSupplier |
+      | manufacturers | graphicCorner                       |
+
+  Scenario: Provide restrictions with empty list of rules
+    When I add a restriction for cart rule rule_free_shipping_1, which requires at least 1 product in cart matching one of these rules:
+      | type     | references |
+      | products |            |
+    Then I should get cart rule error about "empty restriction rule ids"
+    And cart rule rule_free_shipping_1 should have no product restriction rules
+
+  Scenario: Rule to require any quantity of product matching certain rules
+    When I add a restriction for cart rule rule_free_shipping_1, which requires any quantity of product in cart matching one of these rules:
+      | type       | references  |
+      | products   | product1    |
+      | categories | clothes,men |
+    And I add a restriction for cart rule rule_free_shipping_1, which requires any quantity of product in cart matching one of these rules:
+      | type      | references                          |
+      | suppliers | fashionSupplier,accessoriesSupplier |
+    And I add a restriction for cart rule rule_free_shipping_1, which requires at least 100 product in cart matching one of these rules:
+      | type          | references    |
+      | manufacturers | graphicCorner |
+    And I save all the restrictions for cart rule rule_free_shipping_1
+    Then cart rule "rule_free_shipping_1" should have the following product restriction rule groups:
+      | groupReference | quantity | rules count |
+      | free_nr_1      | 0        | 2           |
+      | free_nr_2      | 0        | 1           |
+      | free_nr_3      | 100      | 1           |
+    And the cart rule restriction group "free_nr_1" should have the following rules:
+      | type       | references  |
+      | products   | product1    |
+      | categories | clothes,men |
+    And the cart rule restriction group "free_nr_2" should have the following rules:
+      | type      | references                          |
+      | suppliers | fashionSupplier,accessoriesSupplier |
+    And the cart rule restriction group "free_nr_3" should have the following rules:
+      | type          | references    |
+      | manufacturers | graphicCorner |
+
+  Scenario: Provide restriction group with empty list of restrictions
+    When I add a restriction for cart rule rule_50_percent, which requires at least 1 product, but I provide empty list of rules
+    Then I should get cart rule error about "empty restriction rules"
+    And cart rule rule_50_percent should have no product restriction rules
```

Comment:
SHould we add some scenarios that edit the existing restrictions after initial one have been added? I believe since we're dealing with new cart rules and new group in each scenario you do edit them once but never twice 

And especially we should test that we can remove the previous rules, even remove them all and check that no restrictions are set anymore

Reply:
added couple steps here. Should cover it.
https://github.com/PrestaShop/PrestaShop/pull/32556/commits/781363bf56e65d1d1474debf05563de354f0dbe2